--- src/TerrariaNetCore/Terraria/GameContent/UI/BigProgressBar/PirateShipBigProgressBar.cs
+++ src/tModLoader/Terraria/GameContent/UI/BigProgressBar/PirateShipBigProgressBar.cs
@@ -18,7 +_,7 @@
 		}
 
 		public bool ValidateAndCollectNecessaryInfo(ref BigProgressBarInfo info) {
-			if (info.npcIndexToAimAt < 0 || info.npcIndexToAimAt > 200)
+			if (info.npcIndexToAimAt < 0 || info.npcIndexToAimAt > Main.maxNPCs) // OR==?
 				return false;
 
 			NPC nPC = Main.npc[info.npcIndexToAimAt];
@@ -54,7 +_,7 @@
 		}
 
 		private bool TryFindingAnotherPirateShipPiece(ref BigProgressBarInfo info) {
-			for (int i = 0; i < 200; i++) {
+			for (int i = 0; i < Main.maxNPCs; i++) {
 				NPC nPC = Main.npc[i];
 				if (nPC.active && ValidIds.Contains(nPC.type)) {
 					info.npcIndexToAimAt = i;
